Assets/HoloToolkit/Input/Tests/Scripts/SphereGlobalKeywords.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule.Tests
+{
+    public class SphereGlobalKeywords : MonoBehaviour, ISpeechHandler
+    {
+        private MaterialPropertyBlock propertyBlock;
+
+        private void Start()
+        {
+            propertyBlock = new MaterialPropertyBlock();
+        }
+
+        public void OnSpeechKeywordRecognized(SpeechKeywordRecognizedEventData eventData)
+        {
+            switch (eventData.RecognizedText.ToLower())
+            {
+                case "reset all":
+                    foreach (Renderer renderer in GetComponentsInChildren<Renderer>())
```

Comment:
It's probably best to cache all the renderers here, and be sure to check/update the cache if children are added/removed.
```
private MaterialPropertyBlock propertyBlock;
private Renderer[] childRenderCache;

private void Start()
{
    propertyBlock = new MaterialPropertyBlock();
    childrenRenderCache = GetComponentsInChildren<Renderer>()
}

public void OnSpeechKeywordRecognized(SpeechKeywordRecognixedEventData eventData)
{
    switch(keywords)
    {
        case "reset all":
            for(int i=0; i < childrenRenderCache.Length; i++)
            {
                ChangeColor(childrenRenderCache[i], Color.white);
            {
            break;
    }
}

private void ChangeColor(Renderer, Color color)
{
    // Do Color stuff
}
```